src/main/java/store/service/PaymentService.java

```diff
@@ -0,0 +1,75 @@
+package store.service;
+
+import store.enums.Membership;
+import store.model.Payment;
+import store.model.Product;
+import store.model.Receipt;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PaymentService {
+    public Payment createPayment(List<Product> products, Membership membership) {
+        List<Product> totalProducts = findTotalProducts(products);
+        List<Product> giftProducts = findGiftProducts(products);
+
+        Payment payment = new Payment();
+        payment.setTotalProductsAmount(totalProducts);
+        payment.setPromotionDiscount(giftProducts);
+        payment.setMembershipDiscount(totalProducts, membership);
+        payment.setTotalPaymentAmount();
+
+        return payment;
```

Comment:
아래와 같이 메서드 분리하면 가독성이 더 좋을 것 같아요!

```suggestion
        return createPayment(totalProducts, giftProducts, membership);
```